cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/index.tsx

```diff
@@ -0,0 +1,170 @@
+      marginLeft: '5px',
+      lineHeight: '12px',
+      verticalAlign: 'middle',
+    },
+    widgetContainer: {
+      width: 'calc(100%-1000px)',
+    },
+    widgetDivider: {
+      width: '100%',
+    },
+  };
+};
+
+interface IWidgetCollectionProps extends WithStyles<typeof styles>, ICreateContext {
+  groupID: number;
+}
+
+const WidgetCollectionView: React.FC<IWidgetCollectionProps> = ({
+  classes,
+  groupID,
+  widgetToInfo,
+  groupToWidgets,
+  setWidgetToInfo,
+  setGroupToWidgets,
+}) => {
+  const [activeWidgets, setActiveWidgets] = React.useState(groupID ? groupToWidgets[groupID] : []);
+
+  React.useEffect(() => {
+    if (groupID) {
+      setActiveWidgets(groupToWidgets[groupID]);
+    } else {
+      setActiveWidgets([]);
+    }
+  }, [groupToWidgets]);
+
+  function addWidgetToGroup(index: number) {
+    return () => {
+      const newWidgetID = 'Widget_' + uuidV4();
+
+      // Add a new widget's ID at the specified index
+      const newWidgets = [...activeWidgets];
+      if (newWidgets.length === 0) {
+        newWidgets.splice(0, 0, newWidgetID);
+      } else {
+        newWidgets.splice(index + 1, 0, newWidgetID);
+      }
+      setGroupToWidgets({
+        ...groupToWidgets,
+        [groupID]: newWidgets,
+      });
+
+      // Set the default empty properties of the widget
+      setWidgetToInfo({
+        ...widgetToInfo,
+        [newWidgetID]: {
+          widgetType: '',
+          label: '',
+          name: '',
+        } as IWidgetInfo,
+      });
+    };
+  }
+
+  function deleteWidgetFromGroup(widgetIndex) {
+    return () => {
+      // Grab the widget ID to delete
+      const widgetToDelete = activeWidgets[widgetIndex];
+
+      const newWidgets = [...activeWidgets];
+      newWidgets.splice(widgetIndex, 1);
+      setGroupToWidgets({
+        ...groupToWidgets,
+        [groupID]: newWidgets,
+      });
+
+      const { [widgetToDelete]: info, ...restWidgetToInfo } = widgetToInfo;
+      setWidgetToInfo(restWidgetToInfo);
+    };
+  }
+
+  return (
+    <div className={classes.nestedWidgets} data-cy="widget-wrapper-container">
+      <div className={classes.addWidgetLabel}>
+        Add Widgets
+        <span className={classes.required}>*</span>
+      </div>
+      <div className={classes.widgetContainer}>
+        {activeWidgets.map((widgetID, widgetIndex) => {
+          return (
+            <If condition={widgetToInfo[widgetID]}>
+              <div className={classes.eachWidget}>
+                <WidgetInput
+                  widgetToInfo={widgetToInfo}
+                  widgetID={widgetID}
+                  setWidgetToInfo={setWidgetToInfo}
+                />
+                <WidgetActionButtons
+                  onAddWidgetToGroup={addWidgetToGroup(widgetIndex)}
+                  onDeleteWidgetFromGroup={deleteWidgetFromGroup(widgetIndex)}
+                />
+              </div>
+              <If condition={activeWidgets && widgetIndex < activeWidgets.length - 1}>
```

Comment:
Because I have deleted `<If condition={widgetToInfo[widgetID]}>` in later PR (as noted in the comment above https://github.com/cdapio/cdap/pull/12211/files/aa90989dabe4b1cff0944141cabee82f55d5a46a#r439463982), we no longer have nested `If component.